ONSdigital/census-rm-case-processor
# Update case event history

Do not merge yet. Changes are dependent on 
* [action-scheduler](https://github.com/ONSdigital/census-rm-action-scheduler/pull/18) 
* [case-api](https://github.com/ONSdigital/census-rm-case-api/pull/13)

The above changes have been merged in to this PR, so shouldn't have an impact if this PR is merged in first. 

# Motivation and Context

The Action Scheduler will be sending events whenever a case is selected to be printed. We need to log an event with the Pack Code of the print file.

# What has changed



Please see [this commit](https://github.com/ONSdigital/census-rm-case-processor/pull/27/commits/96ae48e66db324de397d7f4955c4172f4e81de2d) for changes. We have: 
* Changed the persistence model to allow events to be logged at Case level.
* Added a subscription to a new queue from Action Scheduler
* Added a new event type `PRINT_CASE_SELECTED` 
* Added a new event payload DTO called PrintCaseSelected
* Added a message handler to log the event

# How to test?



Follow instructions to run acceptance tests from [this branch](https://github.com/ONSdigital/census-rm-acceptance-tests/pull/83) and make sure they pass

You can also:
1. Run action scheduler on branch `case-history-updates`
1. Load a sample file, make action plan & rule to print a file
1. Check the event log for the cases containing the expected Pack Codes


# Links



https://trello.com/c/stICnQSn/922-rmc-212-case-processor-update-case-history-when-added-to-print-file-13
